### ISSUE TITLE
Can't get network destination pod name from network dashbord

### ISSUE THREAD

**Issue by Kindling-project** [Kindling-project/kindling#44]
**Describe the bug**
Hi, Kindling guys . In network details dashboard I found my pods' connections  encounter  pkg retransmit, but when i want to know connection's pod details  ,i get the source pod name from the <retransmit tables> while destination pod name is missing. By the way, the network destination must be a pod , cause i get the destination workload name.

**What did you expect to see?**
i want to get the network connection's source pod name and also destination pod name.

**What did you see instead?**
A clear and concise description of what you saw instead.

**Screenshots**
![image](https://user-images.githubusercontent.com/96714582/154421533-479a98ff-3b8c-41ab-8d24-0fe64c82eaba.png)

**Comment by dxsup** [comment]
I found that the field "dst_pod" is missed in this piece of data. This is caused because the collector couldn't get destination NAT IP when looking up the conntrack table.

I have checked the related codes and found a bug that could cause this problem. I'll open a PR to fix this.

**Comment by dxsup** [comment]
I have to reopen this issue as I found another bug that could result in this problem. Currently, we get the conntrack table via Netlink socket, and we set the option `NETLINK_LISTEN_ALL_NSID` when initializing it. But we didn't realize that this option [only works since Linux 4.2](https://man7.org/linux/man-pages/man7/netlink.7.html). So for the kernel version less than 4.2, if the pod is not at the same namespace of the root, we can't get its conntrack flows. This could result in this problem.

And also I found that on nodes with a large number of Netlink conntrack updates, the collector has very high CPU usage. Considering these problems, we want to replace the current implementation with a more complete one.
